tools/shell/shell_runner.cpp

```diff
@@ -123,17 +123,30 @@ int main(int argc, char* argv[]) {
         systemConfig.readOnly = true;
     }
 
+    const std::string historyFile = "history.txt";
     auto pathToHistory = args::get(historyPathFlag);
     if (!pathToHistory.empty() && pathToHistory.back() != '/') {
         pathToHistory += '/';
     }
-    pathToHistory += "history.txt";
+    // If `historyFile` is present in the current directory, use that to preserve backward
+    // compatibility.
+    if (pathToHistory.empty() && !std::filesystem::exists(historyFile)) {
```

Comment:
Need to double check, but I assume we must have similar piece of code somewhere else that's creating the kuzu home directory?

Reply:
Yes in `ExtensionInstaller::installExtension` using `ClientContext.homeDirectory`.

Reply:
Moved the getUserHomeDir logic to ClientContext. The extensions code creates something like `~/.kuzu/<extension>/...` which is independent from the history logic.